### PR TITLE
Avoid unnecessary "overwriting a non-zero byte" warnings

### LINKED ISSUE #897: Avoid unnecessary "overwriting a non-zero byte" warnings from `rgbfix`
- [ ] Don't warn if the non-zero byte being written is the same as the original byte.
- [ ] Add a `-O`/`--overwrite` flag to `rgbfix` to silence all "overwriting a non-zero byte" warnings

The first point would allow header data to be redundantly declared in the assembly files without `rgbfix` complaining. The second point would let you explicitly overwrite non-zero data, such as correcting an overlaid base ROM's checksum.

### PR DESCRIPTION
**Pull request by Rangi42**
- Don't warn if the non-zero byte being written is the same as the original byte
- Add a `-O` / `--overwrite` flag to silence all such warnings

Fixes #897

Also happens to fix the case if the global checksum bytes were non-zero (warnings will now be emitted).

### CODE REVIEW

File: src/fix/main.c [hunk 1]
```diff
@@ -149,12 +150,12 @@ enum MbcType {
 	TPP1_TIMER_MULTIRUMBLE_RUMBLE = 0x107,
 	TPP1_BATTERY = 0x108,
 	TPP1_BATTERY_RUMBLE = 0x109,
-	TPP1_BATTERY_MULTIRUMBLE = 0x10a, // Should not be possible
-	TPP1_BATTERY_MULTIRUMBLE_RUMBLE = 0x10b,
-	TPP1_BATTERY_TIMER = 0x10c,
-	TPP1_BATTERY_TIMER_RUMBLE = 0x10d,
-	TPP1_BATTERY_TIMER_MULTIRUMBLE = 0x10e, // Should not be possible
-	TPP1_BATTERY_TIMER_MULTIRUMBLE_RUMBLE = 0x10f,
+	TPP1_BATTERY_MULTIRUMBLE = 0x10A, // Should not be possible
```
Comment (Rangi42) [inline src/fix/main.c:153]: Most hex constants here were already uppercase.

File: src/fix/main.c [hunk 2]
```diff
@@ -746,6 +747,15 @@ static const uint8_t ninLogo[] = {
 	0xDD, 0xDC, 0x99, 0x9F, 0xBB, 0xB9, 0x33, 0x3E
 };
 
+static const uint8_t trashLogo[] = {
+	0xFF^0xCE, 0xFF^0xED, 0xFF^0x66, 0xFF^0x66, 0xFF^0xCC, 0xFF^0x0D, 0xFF^0x00, 0xFF^0x0B,
```
Comment (Rangi42) [inline src/fix/main.c:751]: `~` instead of `0xFF^` doesn't pass `-Woverflow`.

Comment (ISSOtm) [reply src/fix/main.c:751]: It might with suffix `hh`, but I doubt that'd be worth the effort.

File: src/fix/main.c [hunk 3]
```diff
@@ -870,94 +906,68 @@ static void processFile(int input, int output, char const *name, off_t fileSize)
...
 	}
 
 	// ROM size will be written last, after evaluating the file's size
 
-	if ((cartridgeType & 0xff00) == TPP1) {
-		warnNonZero(rom0, 0x149, 2, "TPP1 identification code");
-		rom0[0x149] = 0xC1;
-		rom0[0x14a] = 0x65;
+	if ((cartridgeType & 0xFF00) == TPP1) {
+		uint8_t tpp1Code[2] = {0xC1, 0x65};
```
Comment (ISSOtm) [inline src/fix/main.c:946]:
```suggestion
		uint8_t const tpp1Code[2] = {0xC1, 0x65};
```

**Review by ISSOtm — CHANGES_REQUESTED** [review]

File: src/fix/main.c (hunk 2 above)
Comment (Rangi42) [reply src/fix/main.c:751]: "`h`" or "`hh`" are not [valid suffixes](https://en.cppreference.com/w/c/language/integer_constant). I thought they might be GNU extensions, but GCC 10.2 doesn't compile them. Also I think that `~ X` would promote X to an integer, as would `~ (uint8_t) X`; you'd need `(uint8_t) ~ X` to not do that.

**Review by ISSOtm — APPROVED** [review]

**Pull request merged**